src/libasr/lsp.cpp

```
@@ -13,6 +14,50 @@
 
 namespace LCompilers {
 
+LSP::DiagnosticSeverity diagnostic_level_to_lsp_severity(diag::Level level)
+{
+    switch (level) {
+    case diag::Level::Error:
+        return LSP::DiagnosticSeverity::Error;
+    case diag::Level::Warning:
+        return LSP::DiagnosticSeverity::Warning;
+    case diag::Level::Note:
+        return LSP::DiagnosticSeverity::Information;
+    case diag::Level::Help:
+        return LSP::DiagnosticSeverity::Hint;
+    default:
+        return LSP::DiagnosticSeverity::Warning;
+    }
+}
+
+LSP::SymbolKind asr_symbol_type_to_lsp_symbol_kind(ASR::symbolType symbol_type)
+{
+    switch (symbol_type) {
+    case ASR::symbolType::Module:
+        return LSP::SymbolKind::Module;
+    case ASR::symbolType::Function:
+        return LSP::SymbolKind::Function;
+    case ASR::symbolType::GenericProcedure:
+        return LSP::SymbolKind::Function;
+    case ASR::symbolType::CustomOperator:
+        return LSP::SymbolKind::Operator;
+    case ASR::symbolType::Struct:
+        return LSP::SymbolKind::Struct;
+    case ASR::symbolType::Enum:
+        return LSP::SymbolKind::Enum;
+    case ASR::symbolType::Variable:
+        return LSP::SymbolKind::Variable;
+    case ASR::symbolType::Class:
+        return LSP::SymbolKind::Class;
+    case ASR::symbolType::ClassProcedure:
+        return LSP::SymbolKind::Method;
+    case ASR::symbolType::Template:
+        return LSP::SymbolKind::TypeParameter;
+    default:
+        return LSP::SymbolKind::Function;
```

Here we should probably throw an exception in default, so that if we add more symbols, it shows the exception instead of silently returning Function.